cirq-core/cirq/qis/clifford_tableau.py

```diff
@@ -176,6 +178,43 @@ def _str_full_(self) -> str:
 
         return string
 
+    def merged_with(self, second: 'CliffordTableau') -> 'CliffordTableau':
+        """Returns a merged CliffordTableau of self tableau and second tableau.
+
+        The meaning of merge two clifford tableau is the corresponding unitary operation
+        of merged clifford tableau is equal to (up to global phase) the composed unitary
+        operation of self tableau and the one of second tableau.
+        """
+        m1 = self.matrix().astype(int)
+        m2 = second.matrix().astype(int)
```

Comment:
why is the astype needed? shouldn't we keep these as the correct type in this class?

Reply:
Hmm, the under Clifford Tableau is always 0, 1. So the underlying datatype is `bool`. (It has been that for long time). Boolean type computation is awkward in numpy, hence it is usually easier to transform to int type and do the mod 2 at last.

Reply:
maybe add a comment?

Reply:
Done

Reply:
Just for my understanding, what is awkward about the boolean computation in numpy in this context?
From a cursory look at the logic below, we seem to be doing a bunch of `+` operations followed by `np.mod(_, 2)`. Wouldn't just using the `^` on the bool numpy arrays be more readable (and maybe also efficient)?